RELEASE.md

```diff
@@ -0,0 +1,83 @@
+# Release Checklist
+
+The following checks should be completed before releasing a new version of the
+Polkadot/Kusama/Westend runtime or client:
```

Comment:
```suggestion
The following checks should be completed before publishing a new release of the
Polkadot/Kusama/Westend runtime or client:
```

Just clarifying the language - It has caused confusion before since we have a draft release in paritytech/polkadot/releases. This checklist refers to the checks we should make before we *publish* the release.